### DISCUSSION TITLE
How to use CF_MODPACK_ZIP correctly when a modpack contains mods with disabled project distribution

### DISCUSSION THREAD

**Discussion by reecejeff** [itzg/docker-minecraft-server#2102] [Q&A] [answered]
Hi

As the title suggests, I am having trouble with creating a server (Forge) with curseforge mods having disabled project distribution and CF_MODPACK_ZIP .

While having the environment variables as follows:

![image](https://user-images.githubusercontent.com/112209802/234819797-26b4bb32-cc9a-4efa-9f17-74446d616f15.png)

I have read the docs, specifically:

"
If the authors of the modpack have disallowed project distribution, then the desired client modpack zip will need to be manually downloaded and made available to the container. The path to that file must be passed to CF_MODPACK_ZIP.
"

If for example, I am trying to add a modpack called "ModPack" and a mod called "ModA" and "ModB" have disabled project distribution, will I need to:

1) Download the "ModA" and "ModB" mods from curseforge
- Place it into a zip folder that the container can see
- Specify the mods zip with CF_MODPACK_ZIP

or 

2) Download the "ModPack" modpack from curseforge
- Place it into a folder that the container can see
-  Specify the modpack zip with CF_MODPACK_ZIP

I have tried different combinations of solution 2) above but it is not adding the missing mods so I am lost.

Questions
- How do I get the missing mods added with CF_MODPACK_ZIP?
- Are my environment variables correct for the criteria I am trying to meet? (Forge server with curseforge modpack)

Thank you for your time!

**Comment by reecejeff** [comment, marked as answer]
AH

It all makes sense now, this is literally **ONLY** if the modpack itself has disabled project distribution. I assumed this would be the same for mods inside the modpack. Perhaps the documentation can add a section to provide more guidance in such a case? (Modpacks containing mods that do not allow project distribution)

**Reply by itzg** [reply]
Both can occur. It has become overwhelming creating a clear runtime error, documenting the normal cases, and now documenting these combinations of complications imposed by mod and modpack authors. PRs are welcome to contribute clarifications to the docs. 

Stay tuned there's some incremental improvements on the way to ease some of the pain, [such as this](https://github.com/itzg/docker-minecraft-server/issues/2067#issuecomment-1522098144)

**Reply by reecejeff** [reply]
Thanks for the reply.

I'll be happy to PR a contribution to the docs, but I am currently stuck on how to move forward with my current issue, once I am done I'll give it a go :)!

**Reply by itzg** [reply]
Ah, I mistook your designated answer as having figured out the solutions by reading the documentation and not finding what you needed. 

There's already a documented solution for your scenario with the non-distributed modpack. Refer to the paragraph that starts at https://github.com/itzg/docker-minecraft-server#:~:text=if%20the%20authors%20of%20the%20modpack "If the authors of the modpack"

As for provided individual mods not allowed for distribution, there's a whole section with several options

https://github.com/itzg/docker-minecraft-server#working-with-mods-and-plugins
